docs/resource-stores.md

```diff
@@ -1,16 +1,95 @@
+* **Add** - Creates a new object (errors if the object exists)
+* **Update** - Updates an existing object (errors if the object doesn't exist)
+* **Upsert** - Updates an existing object, or creates the object if it doesn't exist
+* **UpdateSubresource** - Updates a subresource of the object--this must be done separately from **Update**, which does not update subresources
+* **Delete** - Deletes an existing object (errors if the object doesn't exist)
+* **ForceDelete** - Deletes an existing object, does not error if the object doesn't exist
+* **List** - List all object in a namespace with provided filters. Valid filters are [kubernetes label selectors](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#label-selectors)
+
+It is important to keep in mind that, like a typical key-value store, `Update` overwrites the entire object, so the standard pattern for usage is get-and-update to ensure that you don't erase fields. To update only specific parts of an object, use `Client.Patch`.
+
+An example of using a TypedStore to list all objects that match a selector and then update them:
+```go
+store, err := resource.NewTypedStore[*v1.MyObject](v1.Kind(), clientGenerator)
+if err != nil {
+    panic(err)
+}
+ctx := context.TODO()
+// Get resources of v1.MyObject type in all namespaces that have the labels `environment=dev` and `live=true`
+list, err := store.List(ctx, resource.NamespaceAll, "environment=dev", "live=true")
+if err != nil {
+    panic(err)
+}
+// store.List returns a resource.TypedList[*v1.MyObject], so we can iterate through list.Items 
+// to get the *v1.MyObject type instead of using list.GetItems() to get a list of resource.Object types
+for _, item := range list.Items {
+    // Since we have the actual type, we can directly access fields on it without needing to cast
+    item.ObjectMeta.Labels["live"] = false
+    // The first return of the Update method is the updated object, but we can discard that because we don't care about it
+    if _, err := store.Update(ctx, item.GetStaticMetadata().Identifier(), item); err != nil {
+        panic(err)
+    }
+}
+```
+The most useful aspect of this example is that we can directly work with the returned `*v1.MyObject` instances, instead of a `resource.Object` that is typically returned by a Client.
+
+## Store
+
+`resource.Store` is a generic store that allows for working with any kind, not just a single one like `TypedStore` or `SimpleStore`. This comes with the downside that arguments are returned types now use `resource.Object` instead of concrete types. If you're only working with a single kind, prefer using `resource.TypedStore` (you may want to consider multiple `resource.TypedStore` instances for multiple kinds as well if you find the workflow simpler, as the reasource overhead isn't that signficant, considering that `resource.Store` still utilizes a `resource.Client` instance per kind under the hood). 
```

Comment:
It's a bit strange to compare with `SimpleStore` here, because it's not been described yet. I suggest that we either remove this comparison or put the `SimpleStore` section before the `Store` section.